src/main/java/org/prnhs/javaee/swim/services/ProgramService.java

```diff
@@ -6,40 +6,40 @@
 import org.prnhs.javaee.swim.core.dao.ProgramDao;
 import org.prnhs.javaee.swim.core.entity.Program;
 import org.prnhs.javaee.swim.dto.ProgramDto;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 @Service
 public class ProgramService {
     
+    private static final Logger LOGGER = LoggerFactory.getLogger(ProgramService.class);
+    
     @Autowired
     private ProgramDao dao;
     
     public ProgramDto save(ProgramDto programDto){
-        if(programDto == null){
+        if(programDto.getObjective() == null){
+            LOGGER.debug("Save method called in ProgramService but the program object was null");

```

Comment:
I tested it. It works fine. I changed the condition from "programDto==null" to "programDto.getObjective()==null" because even tho when the programDto is null, the if statement doesnt get executed.(Did a lot of testing still dont know why). So as a solution this was the best I could do